### PR TITLE
Revise sinh

### PR DESCRIPTION
**Pull request by bszmelcz**
### Details:
 - refactor ngraph sinh class
 - add type_prop, backend, visitors tests
 - update spec
 - add rounding for integral type for sinh reference implementation
### Tickets:
 - 37479

### CODE REVIEW

File: docs/ops/arithmetic/Sinh_1.md [hunk 1]
```diff
@@ -6,28 +6,28 @@
...
+\f]
+
+**Attributes**: 
 
     No attributes available.
 
 **Inputs**
 
-* **1**: An tensor of type T. **Required.**
+* **1**: An tensor of type T and arbitrary shape. **Required.**
```
Comment (ggalieroc) [inline docs/ops/arithmetic/Sinh_1.md:21]: An tensor of type T and arbitrary shape. ---> A tensor of type \*T\* and arbitrary shape

File: docs/ops/arithmetic/Sinh_1.md [hunk 2]
```diff
@@ -6,28 +6,28 @@
...
 
 **Outputs**
 
-* **1**: The result of element-wise sinh operation. A tensor of type *T*.
+* **1**: The result of element-wise sinh operation. A tensor of type T and the same shape as input.
```
Comment (ggalieroc) [inline docs/ops/arithmetic/Sinh_1.md:25]: The result of element-wise sinh operation. A tensor of type T and the same shape as input. ----> The result of element-wise \*Sinh\* operation applied to the input tensor. A tensor of type \*T\* and the same shape as the input tensor.

File: docs/ops/arithmetic/Sinh_1.md [hunk 3]
```diff
@@ -6,28 +6,28 @@
...
 
 **Types**
 
 * *T*: any numeric type.
```
Comment (ggalieroc) [inline docs/ops/arithmetic/Sinh_1.md:29]: any numeric type ----> any supported numeric type

File: docs/ops/arithmetic/Sinh_1.md [hunk 4]
```diff
@@ -6,28 +6,28 @@
...
 **Short description**: *Sinh* performs element-wise hyperbolic sine (sinh) operation with given tensor.
 
-**Attributes**:
+**Detailed description**: *Sinh* does the following with the input tensor *a*:
+
+\f[
+a_{i} = sinh(a_{i})
+\f]
+
+**Attributes**:
```
Comment (ggalieroc) [inline docs/ops/arithmetic/Sinh_1.md:15]: \*\*Attributes\*\*:  ---> \*\*Attributes\*\*: \*Sinh\* operation has no attributes.

File: docs/ops/arithmetic/Sinh_1.md [hunk 5]
```diff
@@ -6,28 +6,28 @@
...
 
 * *T*: any numeric type.
 
-*sinh* does the following with the input tensor *a*:
-
-\f[
-a_{i} = sinh(a_{i})
-\f]
-
 **Examples**
```
Comment (ggalieroc) [inline docs/ops/arithmetic/Sinh_1.md:31]: Please consider to address this comment here: https://github.com/openvinotoolkit/openvino/pull/6429#discussion_r660415923

File: docs/ops/arithmetic/Sinh_1.md [hunk 6]
```diff
@@ -6,28 +6,28 @@
 
 **Short description**: *Sinh* performs element-wise hyperbolic sine (sinh) operation with given tensor.
 
-**Attributes**:
+**Detailed description**: *Sinh* does the following with the input tensor *a*:
```
Comment (ggalieroc) [inline docs/ops/arithmetic/Sinh_1.md:9]: Let's keep the same style as for Cos spec.

\*\*Detailed description\*\*: \*Sinh\* performs element-wise hyperbolic sine (sinh) operation on a given input tensor, based on the following mathematical formula:

File: docs/ops/arithmetic/Sinh_1.md [hunk 7]
```diff
@@ -6,28 +6,28 @@
 
 **Short description**: *Sinh* performs element-wise hyperbolic sine (sinh) operation with given tensor.
```
Comment (ggalieroc) [inline docs/ops/arithmetic/Sinh_1.md:7]: with given tensor. ---> on a given input tensor

File: ngraph/test/backend/sinh.in.cpp [hunk 8]
```diff
@@ -20,38 +20,91 @@
...
+
+    vector<float> a{1, 3};
+
+    auto test_case = test::TestCase<TestEngine>(f);
+    test_case.add_input<float>({a});
+    test_case.add_expected_output<float>(shape, {sinhf(sinhf(1)), sinhf(sinhf(3))});
+    test_case.run(DEFAULT_FLOAT_TOLERANCE_BITS + 2);
+}
+
+NGRAPH_TEST(${BACKEND_NAME}, sinh_int32)
```
Comment (ggalieroc) [inline ngraph/test/backend/sinh.in.cpp:92]: Please replace the usage of backend and use TestCase to keep consistency of tests

**Comment by ggalieroc** [comment]
Please add SLT with integer precision. You can do this by including the op in this [list](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/activation.cpp#L67)

**Comment by ggalieroc** [comment]
Also add checks for inputs/outputs in [evaluate ](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/src/op/sinh.cpp#L68) method, similarly as in [here](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/src/op/prelu.cpp#L81)

**Review by ggalieroc — CHANGES_REQUESTED** [review]
Please address the comments left

File: ngraph/core/src/op/sinh.cpp [hunk 9]
```diff
@@ -14,7 +14,7 @@
 using namespace std;
 using namespace ngraph;
 
-constexpr NodeTypeInfo op::Sinh::type_info;
+NGRAPH_RTTI_DEFINITION(op::v0::Sinh, "Sinh", 0);
```
Comment (ggalieroc) [inline ngraph/core/src/op/sinh.cpp:17]:
```suggestion
NGRAPH_RTTI_DEFINITION(op::v0::Sinh, "Sinh", 0, util::UnaryElementwiseArithmetic);
```

**Comment by ggalieroc** [comment]
As for last comment, maybe you could try to clean up some [includes ](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/src/op/sinh.cpp#L7-L8) from op class, it looks like some of them are not needed

File: ngraph/test/backend/sinh.in.cpp [hunk 10]
```diff
@@ -81,30 +81,25 @@ NGRAPH_TEST(${BACKEND_NAME}, sinh_in_place)
...
     test_case.add_expected_output<float>(shape, {sinhf(sinhf(1)), sinhf(sinhf(3))});
     test_case.run(DEFAULT_FLOAT_TOLERANCE_BITS + 2);
 }
 
-NGRAPH_TEST(${BACKEND_NAME}, sinh_int32)
+NGRAPH_TEST(${BACKEND_NAME}, sinh_i32)
 {
     Shape shape{5};
-    auto A = make_shared<op::Parameter>(element::i32, shape);
+    auto A = make_shared<op::Parameter>(element::i32, shape);;
```
Comment (ggalieroc) [inline ngraph/test/backend/sinh.in.cpp:95]: nitpick: remove additional semicolon

**Review by ggalieroc — APPROVED** [review]

**Comment by ggalieroc** [comment]
@lazarevevgeny could you please review the spec part?

**Comment by ggalieroc** [comment]
Add the operator to the [list](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/ie_test_utils/functional_test_utils/layer_tests_summary/utils/constants.py) of trusted ops

**Review by lazarevevgeny — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]
Please fix CI

**Comment by jdanieck** [comment]
@bszmelcz I've blocked this PR till we discuss why you did [changes in python api requirements](https://github.com/openvinotoolkit/openvino/pull/6432/commits/ca6130dea6b8797b55fe32a5f202bcf899f1aafc).

File: ngraph/python/requirements_test.txt [hunk 11]
```diff
@@ -3,8 +3,8 @@ flake8-bugbear==21.4.3
 flake8-comprehensions==3.3.0
 flake8-docstrings==1.6.0
 flake8-quotes==3.2.0
-mypy==0.812
-onnx==1.8.1
+mypy==0.910
+onnx==1.9.0
```
Comment (jdanieck) [inline ngraph/python/requirements_test.txt:7]: You shouldn't do such changes. Let's discuss online.

**Review by jdanieck — CHANGES_REQUESTED** [review]

**Review by jdanieck — APPROVED** [review]
@bszmelcz looks like CI is now failing because of a merge conflict, so please resolve it and we should see green finally :)

**Pull request merged**